packages/core/database/src/query/query-builder.ts

```diff
@@ -581,6 +581,18 @@ const createQueryBuilder = (
         let results = rows;
         if (mapResults && state.type === 'select') {
           results = helpers.fromRow(meta, rows);
+          /**
+           * This is possibly covering up an issue in the query itself
+           * Currently, when we do a query with a join, we use a distinct rather than a groupBy
+           * which results in duplicated data in certain cases. However, it is difficult to determine
+           * which queries should have the groupBy (top level queries) and which ones should not
+           * (such as populate subqueries)
+           */
+          if (this.shouldUseDistinct()) {
+            if (results?.length > 0 && 'id' in results[0]) {
+              results = _.uniqBy('id')(results);
+            }
+          }
```

Comment:
wouldn't this option break pagination of populated relations?

Reply:
I don't think so, pagination queries are already returning the correct counts that conflict with the results returned (ie, if you run the test I added on `develop` you'll see that you get 8 results back but pagination says you got 4). That's why I consider this a bit of a hack that just happens to arrive at the correct place.

I have a PR that I'm still working out some problems with that could fix this at a query level.

Reply:
oh, I see what you mean, alright :) !

Reply:
That said, if you have any ideas for any query that would break, please let me know so I can add an api test for it both here and the other PR

Reply:
Was giving it a thought and I can not think of alternatives,
this looks good enough to me. wdyt if you reference the issue in this comment so we know what is precisely solving?

Reply:
Yeah, sounds good. I probably won't merge either of them until @alexandrebodin is back and has a chance to look. Once he sees the exact cause and source of the bug, he may have an idea to actually fix it in the exact right place. I'm just not familiar enough with the intricate details of the query builder other than what I can see in the code and that we have tests for.